src/festim/initial_condition.py

```diff
@@ -98,3 +106,35 @@ def create_expr_fenics(self, mesh, function_space):
                 self.value(**kwargs),
                 get_interpolation_points(function_space.element),
             )
+
+
+def read_function_from_file(
+    filename: str, name: str, timestamp: int | float, family="P", order: int = 1
+) -> fem.Function:
+    """
+    Read a function from a file
+
+    note::
+        The function is read from a file using adios4dolfinx. For more information
+        see the [adios4dolfinx documentation](https://jsdokken.com/adios4dolfinx/README.html).
+
+    Args:
+        filename (str): the filename
+        name (str): the name of the function
+        timestamp (int, float): the timestamp of the function
+        family (str): the family of the function space
+        order (int): the order of the function space
```

Comment:
already have dynamic type hinting
```suggestion
        filename: the filename
        name: the name of the function
        timestamp: the timestamp of the function
        family: the family of the function space
        order: the order of the function space
```